zkevm-circuits/src/pi_circuit.rs

```diff
@@ -226,19 +226,19 @@ impl<F: Field> SubCircuitConfig<F> for PiCircuitConfig<F> {
         let tx_value = tx_table.value;
         let tag = tx_table.tag;
         let index = tx_table.index;
-        let tx_id_inv = meta.advice_column();
-        let tx_value_inv = meta.advice_column();
-        let tx_id_diff_inv = meta.advice_column();
+        let tx_id_inv = meta.advice_column_in(SecondPhase);
+        let tx_value_inv = meta.advice_column_in(SecondPhase);
+        let tx_id_diff_inv = meta.advice_column_in(SecondPhase);
         // The difference of tx_id of adjacent rows in calldata part of tx table
         // lies in the interval [0, 2^16] if their tx_id both do not equal to zero.
         // We do not use 2^8 for the reason that a large block may have more than
         // 2^8 transfer transactions which have 21000*2^8 (~ 5.376M) gas.
         let fixed_u16 = meta.fixed_column();
-        let calldata_gas_cost = meta.advice_column();
-        let is_final = meta.advice_column();
+        let calldata_gas_cost = meta.advice_column_in(SecondPhase);
+        let is_final = meta.advice_column_in(SecondPhase);
```

Comment:
Why `is_final` has to be `SecondPhase`? At least the name suggests that it only stores a bool.

Reply:
Indeed, I checked and that's the case. No rlc-dependant data is assigned to it.

Reply:
If there's no RLC involved, they should be `FirstPhase` as the values of the witness will be used to compute a commitment that would serve as randomness for the RLCs. 
So we have to include ALL the witness that aren't RLC-related as `FirstPhase`